htlcswitch/mailbox.go

```diff
@@ -703,6 +703,23 @@ func (m *memoryMailBox) FailAdd(pkt *htlcPacket) {
 		pkt.originalOutgoingChanID, m.cfg.shortChanID,
 	)
 
+	// Get the update add, critically failing if we have the wrong type of
+	// htlc message.
+	htlc, ok := pkt.htlc.(*lnwire.UpdateAddHTLC)
```

Comment:
If we don't always want to have to type assert to figure out if there's a point or not, we can thread the option thru within the packet itself as a new attribute.